File: array-string/jump-game/jump-game.go
func canJump(nums []int) bool {
    n := len(nums)
    farthest := 0
    for i := 0; i <= farthest && i < n; i++ {
        if i+nums[i] > farthest {
            farthest = i + nums[i]
        }
        if farthest >= n-1 {
            return true
        }
    }
    return n == 1
}

File: array-string/jump-game/jump-game.ts
function canJump(nums: number[]): boolean {
    let farthest = 0;
    for (let i = 0; i <= farthest && i < nums.length; i++) {
        farthest = Math.max(farthest, i + nums[i]);
        if (farthest >= nums.length - 1) return true;
    }
    return nums.length === 1;
};